apps/ledger-live-mobile/src/screens/SendFunds/DomainServiceRecipientRow.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { memo, useCallback, useEffect, useMemo } from "react";
+import { Linking, Platform, StyleSheet, View } from "react-native";
+import Clipboard from "@react-native-community/clipboard";
+import { useDomain } from "@ledgerhq/domain-service/hooks/index";
+import { isLoaded } from "@ledgerhq/domain-service/hooks/logic";
+import { Transaction } from "@ledgerhq/live-common/generated/types";
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/impl";
+import { Trans } from "react-i18next";
+import RecipientInput from "../../components/RecipientInput";
+import Alert from "../../components/Alert";
+import ExternalLink from "../../components/ExternalLink";
+import { urls } from "../../config/urls";
+import TranslatedError from "../../components/TranslatedError";
+import LText from "../../components/LText";
+import SupportLinkError from "../../components/SupportLinkError";
+
+type Props = {
+  onChangeText: (value: string) => void;
+  value: string;
+  onRecipientFieldFocus: () => void;
+  account: AccountLike;
+  parentAccount?: Account | null;
+  transaction: Transaction;
+  setTransaction: (t: Transaction) => void;
+  warning?: Error | null;
+  error?: Error | null;
+};
+
+const DomainServiceRecipientInput = ({
+  onChangeText,
+  value,
+  onRecipientFieldFocus,
+  transaction,
+  setTransaction,
+  account,
+  parentAccount,
+  warning,
+  error,
+}: Props) => {
+  const bridge = getAccountBridge(account, parentAccount);
+  const domainServiceResponse = useDomain(value, "ens");
+  const ensResolution = useMemo(
+    () =>
+      isLoaded(domainServiceResponse)
+        ? domainServiceResponse.resolutions[0]
+        : null,
+    [domainServiceResponse],
+  );
+
+  const lowerCaseValue = useMemo(() => value.toLowerCase(), [value]);
+  const isForwardResolution = useMemo(
+    () => !!lowerCaseValue && lowerCaseValue === ensResolution?.domain,
+    [ensResolution?.domain, lowerCaseValue],
+  );
+
+  useEffect(() => {
+    const { recipient, recipientDomain } = transaction;
+
+    if (!ensResolution) {
+      // without ENS resolution transaction recipient should always be the user's input
+      const recipientUpdated = recipient === value;
+      // without ENS resolution transaction domain should be undefined
+      const hasDomain = !!recipientDomain;
+      if (!recipientUpdated || hasDomain) {
+        setTransaction(
+          bridge.updateTransaction(transaction, {
+            recipient: value,
+            recipientDomain: undefined,
+          }),
+        );
+      }
+      return;
+    }
+
+    // verify if domain is present in transaction
+    const domainSet = recipientDomain === ensResolution;
```

Comment:
- ENS is not "only" Ethereum, it could resolve every coin possible if configured the right way. It's not in this step though backend wise. 
- In our case, ENS is the only possible solution as it's the only one we *want* to support. Since the product hasn't started working on "what should happen if we had multiple resolutions/collisions", we can't predict what the code should look like when we don't have this very specific use case of ENS. I think we should update the names only once we have multiple domain providers and we have a product decision on what should happen in those cases. 
- `recipientDomain` is generic as we could very well implement it on all chains with different providers. E.g.:
  - Cosmos chains -> Stargaze names, DENS.sh, Starname. 
  - EVM -> Lens Protocol.
  - All chains -> ENS, Unstoppable Domains

Reply:
by Ethereum I meant, ethereum as a family, are you saying the wording will remain as "ens" even for other blockchains?

Reply:
Yes it would, you need to see ENS as a provider and its database is the Ethereum blockchain, but it can resolve anything (again, in theory, right now it's only resolving EVM addresses by backend design). 
So you could type "greweb.eth" and resolve a BTC address for example.

That's why the domain-service is an external lib, as technically most of them are not directly linked to a single family. But as said before I guess the addition of new providers will tell us how we'll have to make that thing evolve 🤞